### PR TITLE
fix: make sure atomic 64bit fields are 64bit aligned

### PR DESCRIPTION
**Pull request by Stebalien**
Otherwise, this won't work on 32bit ARM.

See https://golang.org/pkg/sync/atomic/#pkg-note-BUG

### CODE REVIEW

**Comment by Stebalien** [comment]
Or we can just fix https://github.com/filecoin-project/lotus/issues/5795.

**Review by magik6k — COMMENTED** [review]
Yeah, not sure how much we save by not using a RWMutex, it seems like the cleaner solution

We should also review other uses of atomics in the codebase (maybe check  if there are any linters which could detect this)

**Comment by ribasushi** [comment]
@Stebalien afaik filecoin-ffi has a hard-dependency on 64bit instruction sets, so this is moot in the context of lotus

**Review by magik6k — APPROVED** [review]

**Comment by Stebalien** [comment]
TL;DR: There's no reason _not_ to fix this. While we never plan on supporting 32bit platforms, someone else might come along and do that.

Alternatively, someone might try to create a proof-less "lotus lite", extract this code into a separate package, fork, etc.

**Comment by yaohcn** [comment]
> Yeah, not sure how much we save by not using a RWMutex, it seems like the cleaner solution
> 
> We should also review other uses of atomics in the codebase (maybe check if there are any linters which could detect this)

There are many uses of atomics in the codebase.
